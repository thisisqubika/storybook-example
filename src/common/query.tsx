import {
  QueryClient, QueryClientConfig, QueryClientProvider,
} from '@tanstack/react-query';

export function queryClientConfig(
  config?: QueryClientConfig,
): QueryClientConfig {
  return {
    ...config,
    defaultOptions: {
      ...config?.defaultOptions,
      queries: {
        refetchOnWindowFocus: false,
        retry: 1,
        ...config?.defaultOptions?.queries,
      },
    },
  };
} // function newQueryClient

export interface QueryClientWrapperProps {
  config?: QueryClientConfig;
  children: JSX.Element;
} // interface ThemeWrapperProps

export function QueryClientWrapper({
  config,
  children,
}: QueryClientWrapperProps) {
  const queryClient = new QueryClient(queryClientConfig(config));
  return (
    <QueryClientProvider client={queryClient}>
      {children}
    </QueryClientProvider>
  );
} // function QueryClientWrapper

