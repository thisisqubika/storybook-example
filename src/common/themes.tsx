// Check out <https://zenoo.github.io/mui-theme-creator/>.
import {
  createTheme, CssBaseline, ThemeProvider, ThemeOptions,
} from '@mui/material';

export const themeOptions: ThemeOptions = {
  palette: {
    mode: 'light',
    primary: {
      main: '#812020',
    },
    secondary: {
      main: '#cc8333',
    },
    background: {
      default: '#e0cab8',
      paper: '#efe8e2',
    },
    error: {
      main: '#fb3636',
    },
    info: {
      main: '#025e90',
    },
    warning: {
      main: '#e49e01',
    },
    success: {
      main: '#2b9430',
    },
  },
  typography: {
    fontFamily: 'Lora',
    fontSize: 13,
  },
}; // const themeOptions

export interface ThemeWrapperProps {
  children: JSX.Element;
} // interface ThemeWrapperProps

export function ThemeWrapper({
  children,
}: ThemeWrapperProps) {
  return (
    <ThemeProvider theme={createTheme(themeOptions)}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  );
} // function ThemeWrapper
