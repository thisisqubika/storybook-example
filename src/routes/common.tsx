import { isRouteErrorResponse, useRouteError } from 'react-router-dom';

export const PATHS = {
  books: 'books',
}; // const PATHS

export function ErrorBoundary() {
  const error: unknown = useRouteError();
  return isRouteErrorResponse(error) ? (
    <h1>{error.status} {error.statusText}</h1>
  ) : (
    <h1>{(error as Error)?.message ?? error}</h1>
  );
} // function ErrorBoundary
