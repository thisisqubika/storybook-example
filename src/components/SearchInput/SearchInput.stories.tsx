import type { Meta, StoryObj } from '@storybook/react';
import { expect, fn, userEvent, within } from '@storybook/test';
import { ThemeWrapper } from '../../common/themes';
import { SearchInput, type SearchInputProps } from '.';

export default {
  component: SearchInput,
} satisfies Meta<SearchInputProps>; // meta

export const Base: StoryObj<SearchInputProps> = {
  args: {
    color: 'black',
    disabled: false,
    initialTerm: '',
    onSearch: fn(),
  } satisfies SearchInputProps,

  render(props: SearchInputProps) {
    return (
      <ThemeWrapper>
        <SearchInput {...props} />
      </ThemeWrapper>
    );
  }, // render

  async play({ args, canvasElement }) {
    const canvas = within(canvasElement);
    const searchInput = canvas.getByTestId('SearchInput.input');
    const testInputText = 'Book search';
    await userEvent.type(searchInput, testInputText, { delay: 100 });
    const iconButton = canvas.getByRole('button');
    await userEvent.click(iconButton);
    await expect(args.onSearch).toHaveBeenCalledWith(testInputText);
  }, // play

}; // const Base
