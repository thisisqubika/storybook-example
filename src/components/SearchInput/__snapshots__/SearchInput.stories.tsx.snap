// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SearchInput Base smoke-test 1`] = `
<div class="css-r7zuen">
  <div class="MuiFormControl-root MuiTextField-root css-136h9lv-MuiFormControl-root-MuiTextField-root"
       data-testid="SearchInput"
  >
    <div class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1lohisl-MuiInputBase-root-MuiInput-root">
      <input aria-invalid="false"
             id=":r0:"
             placeholder="Search..."
             type="text"
             data-testid="SearchInput.input"
             class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
             value
      >
      <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root">
        <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1m1cnyu-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
                data-testid="SearchInput.IconButton"
        >
          <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-xhi8ul-MuiSvgIcon-root"
               focusable="false"
               aria-hidden="true"
               viewbox="0 0 24 24"
               data-testid="SearchIcon"
          >
            <path d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14">
            </path>
          </svg>
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
